x-pack/plugins/security_solution/common/detection_engine/schemas/alerts/8.8.0/index.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ALERT_URL } from '@kbn/rule-data-utils';
+import type { AlertWithCommonFields800 } from '@kbn/rule-registry-plugin/common/schemas/8.0.0';
+import type {
+  BaseFields840,
+  EqlBuildingBlockFields840,
+  EqlShellFields840,
+  NewTermsFields840,
+} from '../8.4.0';
+
+/* DO NOT MODIFY THIS SCHEMA TO ADD NEW FIELDS. These types represent the alerts that shipped in 8.6.0.
+Any changes to these types should be bug fixes so the types more accurately represent the alerts from 8.6.0.
+If you are adding new fields for a new release of Kibana, create a new sibling folder to this one
+for the version to be released and add the field(s) to the schema in that folder.
+Then, update `../index.ts` to import from the new folder that has the latest schemas, add the
+new schemas to the union of all alert schemas, and re-export the new schemas as the `*Latest` schemas.
+*/
+
+export type GenericAlert880 = AlertWithCommonFields800<
+  BaseFields840 & { [ALERT_URL]: string | undefined }
```

Comment:
Since `ALERT_URL` is common to all the alert types, can you define it in a new interface `BaseFields880` in this file and have the rule-specific alert interfaces extend `BaseFields880` instead of the 8.4 versions of those interfaces?

Reply:
Also, originally when adding these schemas I was torn between whether we should extend prior versions when adding new fields (avoiding duplication but adding some complexity) or simply copy the old version and add the new fields. I think with the way the schemas have evolved it might be easier to read if we copy the previous version and accept the duplication of existing fields in the new schema.

I imagine labeling each field with when it was added could help with the duplication, e.g.
```
export interface BaseFields880 {
  [TIMESTAMP]: string; // ADDED IN: 8.0.0
  [SPACE_IDS]: string[]; // ADDED IN: 8.0.0
  [EVENT_KIND]: 'signal'; // ADDED IN: 8.0.0
  ...
  [ALERT_RULE_VERSION]: number; // ADDED IN: 8.0.0
  'kibana.alert.rule.risk_score': number; // ADDED IN: 8.0.0
  'kibana.alert.rule.severity': string; // ADDED IN: 8.0.0
  'kibana.alert.rule.building_block_type': string | undefined; // ADDED IN: 8.0.0
  [ALERT_RULE_INDICES]: string[]; // ADDED IN: 8.4.0
  [ALERT_URL]: string | undefined; // ADDED IN: 8.8.0
  [key: string]: SearchTypes; // ADDED IN: 8.0.0
}
```

For this PR it's completely fine to continue extending old versions of the interfaces, but I wanted to note that we don't have to stick with that design going forward and if it makes more sense to you to copy all the fields over then that's a reasonable approach too (and one we may switch to in the future).

Reply:
Yea, I wanted to do that, but since it doesn't look like the UUID is generated at that point, it made it a bit harder since the url depends on that. Since I didn't have access to the `ALERT_UUID` in `build_alert.ts`, I just followed the same pattern as how `ALERT_UUID` was set in the 8.4.0 file.

Reply:
https://github.com/elastic/kibana/pull/155069/commits/c916500cf6a7b8814de3467d6125210857e82ee8